File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/api/donations/PayPalCreatePaymentMethodPayload.java
/*
 * Copyright 2025 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.api.donations;

import com.fasterxml.jackson.annotation.JsonProperty;

class PayPalCreatePaymentMethodPayload {
  @JsonProperty
  private String returnUrl;

  @JsonProperty
  private String cancelUrl;

  PayPalCreatePaymentMethodPayload(String returnUrl, String cancelUrl) {
    this.returnUrl = returnUrl;
    this.cancelUrl = cancelUrl;
  }
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/api/donations/PayPalCreatePaymentMethodPayload.kt
/*
 * Copyright 2025 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.api.donations

/**
 * Request JSON for creating a recurring PayPal payment method
 */
internal class PayPalCreatePaymentMethodPayload(
  val returnUrl: String,
  val cancelUrl: String
)
